src/ClientData/TimerDataTest.cpp

```diff
@@ -226,4 +226,77 @@ TEST(TimerData, GetTimers) {
   CheckGetTimers(GetTimersDifferentDepths());
 }
 
+TEST(TimerData, GetTimersAtDepthDiscretized) {
+  // Left, right and down timers
+  std::unique_ptr<TimerData> timer_data = GetTimersDifferentDepths();
+
+  uint32_t kOnePixel = 1;
+  uint32_t kNormalResolution = 1000;
+
+  // Normal case. Left and right timer are visible.
+  EXPECT_EQ(
+      timer_data->GetTimersAtDepthDiscretized(0, kNormalResolution, kLeftTimerStart, kRightTimerEnd)
+          .size(),
+      2);
+
+  // Range tests.
+  {
+    // No visible timers at the left and right of the visible range.
+    EXPECT_EQ(timer_data->GetTimersAtDepthDiscretized(0, kNormalResolution, 0, kLeftTimerStart - 1)
+                  .size(),
+              0);
+    EXPECT_EQ(timer_data
+                  ->GetTimersAtDepthDiscretized(0, kNormalResolution, kRightTimerEnd + 1,
+                                                kRightTimerEnd + 10)
+                  .size(),
+              0);
+
+    // Only left timer will be visible if the right timer is out of range.
+    EXPECT_EQ(timer_data
+                  ->GetTimersAtDepthDiscretized(0, kNormalResolution, kLeftTimerStart,
+                                                kRightTimerStart - 1)
+                  .size(),
+              1);
+
+    // Only right timer will be visible if the left timer is out of range.
+    EXPECT_EQ(
+        timer_data
+            ->GetTimersAtDepthDiscretized(0, kNormalResolution, kLeftTimerEnd + 1, kRightTimerEnd)
+            .size(),
+        1);
+
+    // Both timers will be visible even if we include them partially.
+    EXPECT_EQ(
+        timer_data
+            ->GetTimersAtDepthDiscretized(0, kNormalResolution, kLeftTimerEnd, kRightTimerStart)
+            .size(),
+        2);
+  }
+
+  // Resolution tests.
+  {
+    // Only one timer will be visible if we have 1 pixel resolution.
+    EXPECT_EQ(timer_data->GetTimersAtDepthDiscretized(0, kOnePixel, kLeftTimerStart, kRightTimerEnd)
+                  .size(),
+              1);
+
+    // Only one timer will be visible if we zoom-out a lot even with a normal resolution.
+    EXPECT_EQ(timer_data->GetTimersAtDepthDiscretized(0, kNormalResolution, 0, 10000000).size(), 1);
+
+    // If there is a timer in the range, we should see it in any resolution.
+    EXPECT_EQ(
+        timer_data->GetTimersAtDepthDiscretized(0, kOnePixel, kLeftTimerStart, kLeftTimerStart + 1)
+            .size(),
+        1);
+    EXPECT_EQ(timer_data
+                  ->GetTimersAtDepthDiscretized(0, kNormalResolution, kLeftTimerStart,
+                                                kLeftTimerStart + 1)
+                  .size(),
+              1);
+  }
+
+  // Down timer
+  EXPECT_EQ(timer_data->GetTimersAtDepthDiscretized(/*depth=*/1, 1000, 0, 1000).size(), 1);
```

Comment:
```suggestion
  EXPECT_EQ(timer_data->GetTimersAtDepthDiscretized(/*depth=*/1, kNormalResolution, 0, 1000).size(), 1);
```

Also either, remove the /*depth=*/ or add it everywhere else. Same for the start and end.